### PR TITLE
Use Box.objects.get_or_create

### PR DESCRIPTION
**Pull request by jaap3**
This makes it possible to create a release without installing the Box fixtures.

This should be perfectly safe as the box content is set immediately after getting/creating the box.

### CODE REVIEW

**Comment by jaap3** [comment]
Looking at this again I think this could even be changed to `update_or_create`. Let me know if that's preferred.

**Review by berkerpeksag — COMMENTED** [review]
You're right, changing the code to use `update_or_create()` would be better.

**Comment by jaap3** [comment]
I've updated the PR to use `update_or_create`, I've also changed the use of `Box.objects.get` / `Box.objects.get_or_create` in other places

File: blogs/parser.py [hunk 1]
```diff
@@ -40,6 +40,10 @@ def update_blog_supernav():
...
     rendered_box = _render_blog_supernav(latest_entry)
 
-    box = Box.objects.get(label='supernav-python-blog')
-    box.content = rendered_box
-    box.save()
+    box, _ = Box.objects.update_or_create(
+        label='supernav-python-blog',
+        defaults={
+            'content': rendered_box,
+            'content_markup_type': 'html'
```
Comment (berkerpeksag) [inline blogs/parser.py]: Nit: Please add a trailing comma.

File: downloads/models.py [hunk 2]
```diff
@@ -158,9 +158,13 @@ def update_supernav():
...
     })
 
-    box = Box.objects.get(label='supernav-python-downloads')
-    box.content = content
-    box.save()
+    box, _ = Box.objects.update_or_create(
+        label='supernav-python-downloads',
+        defaults={
+            'content': content,
+            'content_markup_type': 'html'
```
Comment (berkerpeksag) [inline downloads/models.py]: Nit: Please add a trailing comma.

File: downloads/models.py [hunk 3]
```diff
@@ -173,13 +177,17 @@ def update_supernav():
...
         'latest_python2_source': latest_python2_source,
         'latest_python3_source': latest_python3_source,
     })
-    source_box.content = source_content
-    source_box.save()
+    source_box, _ = Box.objects.update_or_create(
+        label='download-sources',
+        defaults={
+            'content': source_content,
+            'content_markup_type': 'html'
```
Comment (berkerpeksag) [inline downloads/models.py]: Nit: Please add a trailing comma.

File: downloads/models.py [hunk 4]
```diff
@@ -198,9 +206,13 @@ def update_homepage_download_box():
...
     })
 
-    box = Box.objects.get(label='homepage-downloads')
-    box.content = content
-    box.save()
+    box, _ = Box.objects.update_or_create(
+        label='homepage-downloads',
+        defaults={
+            'content': content,
+            'content_markup_type': 'html'
```
Comment (berkerpeksag) [inline downloads/models.py]: Nit: Please add a trailing comma.

File: successstories/models.py [hunk 5]
```diff
@@ -100,9 +100,13 @@ def update_successstories_supernav(sender, instance, created, **kwargs):
...
         })
 
-        box, _ = Box.objects.get_or_create(label='supernav-python-success-stories')
-        box.content = content
-        box.save()
+        box, _ = Box.objects.update_or_create(
+            label='supernav-python-success-stories',
+            defaults={
+                'content': content,
+                'content_markup_type': 'html'
```
Comment (berkerpeksag) [inline successstories/models.py]: Nit: Please add a trailing comma.

**Review by berkerpeksag — COMMENTED** [review]
This looks like a good cleanup, thanks! I only have a comment about using trailing commas after the last item of the dict.

**Comment by jaap3** [comment]
I've added the trailing commas. 

On a related note: It might be interesting to run [black](https://black.readthedocs.io/en/stable/) on this codebase at some point. That would fix a lot of existing code style inconsistencies.

**Review by berkerpeksag — APPROVED** [review]

**Comment by berkerpeksag** [comment]
Thanks!

> On a related note: It might be interesting to run black on this codebase at some point. That would fix a lot of existing code style inconsistencies.

Sorry, I dislike some of the style choices of black :) Also, we don't accept cosmetic-only patches in CPython, so I'd like to follow that rule in python.org as well.

**Pull request merged**